scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -3409,9 +3451,76 @@ def __get_log_config_for_container(self, cid, info, k8s_cache, base_attributes):
         if "parser" in attrs:
             result["parser"] = attrs["parser"]
 
-        return result
+        results = [result]
+
+        # Based on the pod annotations in a format {container_name}.{team}.secret={secret_name}
+        # we might want to add api_keys parameter
+        if container_annotations or all_annotations:
+            api_keys = self.__container_api_keys_from_annotations(k8s_cache, pod_namespace, pod_name, container_annotations, all_annotations)
+
+            if api_keys:
+                results = [
+                    {**result, "api_key": api_key}
+                    for api_key in list(filter(lambda api_key: api_key is not None, api_keys))
+                ]
```

Comment:
This is a little subtle so I would consider adding a comment here about this, something like "Multiple matching api keys will result in multiple log configs"